File: src/services/Credito/SB.Credito.Domain/Entities/Credito.cs
using SB.Core.Domain;
using SB.Core.ValueObjects;
using System;

namespace SB.Credito.Domain.Entities
{
    public class Credito : Entity, IAggregateRoot
    {
        public Credito(int usuarioCod,
                       string usuarioNome,
                       string cpf,
                       DateTime dataNascimento,
                       decimal salario,
                       string observacao)
        {
            UsuarioCod = usuarioCod;
            UsuarioNome = usuarioNome;
            Cpf = new CPF(cpf);
            DataNascimento = dataNascimento;
            Salario = salario;
            Observacao = observacao;
        }

        protected Credito() { }

        public int UsuarioCod { get; set; }
        public string UsuarioNome { get; private set; }
        public CPF Cpf { get; private set; }
        public DateTime DataNascimento { get; private set; }
        public decimal Salario { get; private set; }
        public string Observacao { get; private set; }
    }
}

File: src/web/credito.spa/src/app/@core/models/credito.ts
export class Credito {
  id: string;
  usuarioCod: number;
  usuarioNome: string;
  cpf: string;
  dataNascimento: string;
  salario: number;
  observacao: string;

  constructor(
    id: string = "",
    usuarioCod: number,
    usuarioNome: string,
    cpf: string,
    dataNascimento: string,
    salario: number,
    observacao: string
  ) {
    this.id = id;
    this.usuarioCod = usuarioCod;
    this.usuarioNome = usuarioNome;
    this.cpf = cpf;
    this.dataNascimento = dataNascimento;
    this.salario = salario;
    this.observacao = observacao;
  }

  isValid(): Array<Error> {
    var errors = new Array<Error>();

    if (!this.usuarioNome)
      errors.push(new Error("O nome do cliente deve ser preenchido"));

    if (!this.cpf) errors.push(new Error("O cpf deve ser preenchido"));

    if (!this.dataNascimento)
      errors.push(new Error("A data de nascimento deve ser preenchido"));

    if (!this.isValidDate(this.dataNascimento))
      errors.push(
        new Error("A data da movimentação está em um formato inválido")
      );

    if (!this.observacao)
      errors.push(new Error("O código do tributo deve ser preenchido"));

    return errors;
  }

  isValidDate(data: string): boolean {
    return new Date(data).toUTCString() !== "Invalid Date";
  }
}
